.github/workflows/lint-test-sdk.yml

```diff
@@ -1,13 +1,33 @@
 name: Lint and Test SDK
+
+env:
+  SDK_BRANCH_NAME: ${{ inputs.sdk_branch  || github.head_ref || github.ref_name || 'main' }}
```

Comment:
allow `workflow_call` to spec an SDK branch OR use _head_ref_ or _ref_name_ which are github variables set to point to the local branch this workflow is running on (ex: a PR branch)